error: cannot convert 'struct string' to 'struct array'

**V version:** V 0.2.2 0da827f.8caabf0
**OS:** linux, Debian GNU/Linux bullseye/sid

**What did you do?**
This is the minimum example I could reproduce this with, you can also find it over at https://codeberg.org/eventbike/vlang_plg/src/branch/main/2021-04-08_cannot_describe
~~~
module main

import vweb

pub struct Project {
	vweb.Context
}

fn main() {
	vweb.run<Project>(9080)
}

struct ExampleStruct {
	example int
}

pub fn (mut app Project) request_raw_2() vweb.Result {
	stuff := []ExampleStruct{}
	return app.request_raw(stuff)
}

pub fn (mut app Project) request_raw(foo []ExampleStruct) vweb.Result {
	return $vweb.html()
}

~~~

**What did you expect to see?**
request_raw_2 can be used to return request_raw with some different properties
 
**What did you see instead?**
~~~
/tmp/v/2021-04-08_cannot_describe.5642974674744139927.tmp.c:21676: error: cannot convert 'struct string' to 'struct array'
builder error: 
==================
C error. This should never happen.
~~~

--- a/vlib/v/gen/c/comptime.v
+++ b/vlib/v/gen/c/comptime.v
@@ -477,7 +477,8 @@ fn (mut g Gen) comptime_for(node ast.ComptimeFor) {
 				rec_sym := g.table.sym(method.receiver_type)
 				if rec_sym.kind == .struct_ {
 					if _ := g.table.find_field_with_embeds(rec_sym, 'Context') {
-						if method.generic_names.len > 0 {
+						if method.generic_names.len > 0
+							|| (method.params.len > 1 && method.attrs.len == 0) {
 							continue
 						}
 					}

--- a/vlib/vweb/vweb_app_test.v
+++ b/vlib/vweb/vweb_app_test.v
@@ -92,3 +92,17 @@ fn (mut app App) some_helper<T>(result T) ApiSuccessResponse<T> {
 fn (mut app App) ok() vweb.Result {
 	return app.json(app.some_helper(123))
 }
+
+struct ExampleStruct {
+	example int
+}
+
+fn (mut app App) request_raw_2() vweb.Result {
+	stuff := []ExampleStruct{}
+	return app.request_raw(stuff)
+}
+
+// should compile, this is a helper method, not exposed as a route
+fn (mut app App) request_raw(foo []ExampleStruct) vweb.Result {
+	return app.text('Hello world')
+}

